catalog/core/catalog-core-transform-impl/src/test/java/org/codice/ddf/catalog/transform/impl/TransformImplTest.java

```diff
@@ -0,0 +1,497 @@
+  }
+
+  @Test
+  public void testTransformMetacardsByTransformerIdOnly()
+      throws CatalogTransformerException, MimeTypeParseException {
+
+    BinaryContent binaryContent = mock(BinaryContent.class);
+
+    MultiMetacardTransformer transformer = mock(MultiMetacardTransformer.class);
+    when(transformer.transform(any(List.class), any(Map.class)))
+        .thenReturn(Collections.singletonList(binaryContent));
+
+    when(transformerLocator.findMultiMetacardTransformers(eq("xyz")))
+        .thenReturn(Collections.singletonList(transformer));
+
+    List<BinaryContent> binaryContents =
+        transform.transform(Collections.singletonList(metacard), "xyz", Collections.emptyMap());
+
+    assertThat(binaryContents, hasSize(1));
+    assertThat(binaryContents.get(0), is(binaryContent));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testTransformMetacardsByTransformerIdOnlyEmptyMetacards()
+      throws CatalogTransformerException, MimeTypeParseException {
+
+    transform.transform(Collections.emptyList(), "xyz", Collections.emptyMap());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testTransformMetacardsByTransformerIdOnlyEmptyTransformerId()
+      throws CatalogTransformerException, MimeTypeParseException {
+
+    transform.transform(Collections.singletonList(metacard), null, Collections.emptyMap());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testTransformMetacardsByTransformerIdOnlyTransformerNotFound()
+      throws CatalogTransformerException, MimeTypeParseException {
+
+    when(transformerLocator.findMultiMetacardTransformers(eq("xyz")))
+        .thenReturn(Collections.emptyList());
+
+    transform.transform(Collections.singletonList(metacard), "xyz", Collections.emptyMap());
+  }
+
+  @Test
+  public void testTransformMetacardsByTransformerIdOnlyOneFailOneSuccess()
+      throws CatalogTransformerException, MimeTypeParseException {
+
+    BinaryContent binaryContent = mock(BinaryContent.class);
+
+    MultiMetacardTransformer transformer1 = mock(MultiMetacardTransformer.class);
+    when(transformer1.transform(any(List.class), any(Map.class)))
+        .thenThrow(CatalogTransformerException.class);
+
+    MultiMetacardTransformer transformer2 = mock(MultiMetacardTransformer.class);
+    when(transformer2.transform(any(List.class), any(Map.class)))
+        .thenReturn(Collections.singletonList(binaryContent));
+
+    when(transformerLocator.findMultiMetacardTransformers(eq("xyz")))
+        .thenReturn(Arrays.asList(transformer1, transformer2));
+
+    List<BinaryContent> binaryContents =
+        transform.transform(Collections.singletonList(metacard), "xyz", Collections.emptyMap());
+
+    assertThat(binaryContents, hasSize(1));
+    assertThat(binaryContents.get(0), is(binaryContent));
+  }
+
+  @Test(expected = CatalogTransformerException.class)
+  public void testTransformMetacardsByTransformerIdOnlyEmptyBinaryContents()
```

Comment:
What does the `OnlyEmptyBinaryContents` part of the test name refer to?

Reply:
Added javadoc to better explain the test case.